### DISCUSSION TITLE
【疑难解决】解决PaddleOCR历史存在的疑难Issue

### DISCUSSION THREAD

**Discussion by Liyulingyue** [PaddlePaddle/PaddleOCR#13057] [Ideas & Features]
The dedicated team from the community picked [long standing issues](https://github.com/PaddlePaddle/PaddleOCR/issues?page=1&q=is%3Aopen+is%3Aissue+label%3Atriaged) , please help to solve, this is a community effort to improve PaddleOCR maintenance. 

# 背景
飞桨套件曾经凭借其出色的性能吸引了诸多的开发者，但随着社区发展，这些套件中的一些历史问题也暴露出来，例如代码Bug没有及时处理、功能缺失、兼容性不足等。以PaddleOCR 项目为例，该套件有非常多的使用者，在 issue 区的讨论也很多。甚至有不少 issue 已经是长期存在的 issue。这些 issue 缺少诊断，复现，以及修复。因此，期望能够挑选部分长期存在的，讨论较多的issue，能够进行分析、复现并解决。

更多关于此Issue的信息可以参考 https://github.com/PaddlePaddle/community/pull/859 。

# 现状
我们对部分Issue进行了统计和分类，根据这些Issue，当前的PaddleOCR Issue中存在的问题主要由以下几部分组成：
1. **多语言OCR开发**：当前有很多用户希望基于他们工作所使用到的语言、他们的母语环境训练特定语言的OCR识别模型，但由于文档缺失/依赖不明确等问题，导致训练工作并不顺利、训练后的模型准确性较低。
2. **兼容性问题**：由于平台、GPU环境、文档缺失、依赖不明确的问题，导致部分用户无法在自己的电脑上运行PaddleOCR，因此需要针对该问题进行完善，例如明确依赖关系、指出不同平台的依赖性。
3. **OCR结果的准确性和数据缺失**：在不同的环境下，训练、导出、推理都可能带来一定的精度损失，部分用户的Issue中表明OCR推理有时候会存在丢失特定区域文字的现象。
4. **参考文档缺失或不明确**：在代码的开发过程中，文档工作没有被很好的跟进，从而导致用户无法获得完整的代码功能说明，并给用户的推理、训练工作带来较大的阻碍。
5. **其他未解决的BUG**

为了促进PaddleOCR的发展，提高框架的稳定性和可维护性，我们希望基于当前的Issue对PaddleOCR的文档、依赖关系、Bug等方面进行完善。欢迎有兴趣的开发者一起参与开发~

# 任务划分
当前计划解决的Issue如下表所示：
Update 2024-4-26: 为了更好地促进PaddleOCR的发展，我们不仅仅针对过往Issue进行解决，还增加了一些新的更新目标，见问题11、12。

|序号|Issue|说明|技术路线|认领人/状态/PR号|
|---|---|---|---|---|
| 1 | https://github.com/PaddlePaddle/PaddleOCR/issues/10760 | 关于模型微调和cuda版本不匹配的问题 | - |  | 
| 2 | https://github.com/PaddlePaddle/PaddleOCR/issues/10685 | 更新Backbone后无法运行，解决了一部分 | 复现此问题，并根据评论区信息跑通代码，完善相关文档 | @mattheliu <img src="https://img.shields.io/badge/状态-完成任务-9B59B6" /> [#12140](https://github.com/PaddlePaddle/PaddleOCR/pull/12140)<br> | 
| 3 | https://github.com/PaddlePaddle/PaddleOCR/issues/10288 | 打包后GPU模式下无法运行 | - |  | 
| 4 | https://github.com/PaddlePaddle/PaddleOCR/issues/10197 | 关于模型自动下载的问题，和解码问题 | - |  | 
| 5 | https://github.com/PaddlePaddle/PaddleOCR/issues/6559 | 内存泄露 | 复现此问题，并验证最新版本的Paddle以及PaddleOCR是否存在此问题，逐渐缩小问题范围 | @GreatV <img src="https://img.shields.io/badge/状态-报名-2ECC71" /> <br> | 
| 6 | https://github.com/PaddlePaddle/PaddleOCR/issues/11149 | 训练问题 | 可以针对此问题增加文档说明 |  | 
| 7 | https://github.com/PaddlePaddle/PaddleOCR/issues/11551 | 导出报错 | - |  | 
| 8 | https://github.com/PaddlePaddle/PaddleOCR/issues/10499 | 文本检测训练完进行单张图片预测时发现漏检内容 | 复现此问题，并排查问题原因，修复相关代码 | @mattheliu <img src="https://img.shields.io/badge/状态-报名-2ECC71" /> <br> | 
| 9 | https://github.com/PaddlePaddle/PaddleOCR/issues/8743 | Code doesn't work with numpy>=1.24 | 基于PaddleOCR的requirements.txt，梳理依赖关系，尽可能减少依赖项，并给出最低依赖版本 | @Liyulingyue <img src="https://img.shields.io/badge/状态-完成任务-9B59B6" /> [#11946](https://github.com/PaddlePaddle/PaddleOCR/pull/11946) [#11947](https://github.com/PaddlePaddle/PaddleOCR/pull/11947)<br> | 
| 10 | https://github.com/PaddlePaddle/PaddleOCR/issues/11441 | 训练时的shape问题，可以增加一下README | - |  | 
| 11 | #10270 | PPStructure版面分析得到的结果，bbox里OCR的结果缺失最后一行 | - | @RussellLuo #11916 | 
|12||setup.py的迁移改造|使用setup对包进行配置已经过时了，需要将配置程序更改为toml|@Liyulingyue|
|13||CI流水线改造||@GreatV|
|14| #11706 #11079 | freeze on mac | using Apple's BLAS implementation to replace openblas on mac | @jzhang533 https://github.com/PaddlePaddle/Paddle/pull/64408|

另：大家也可以从这个更大的池子里来认领： [long standing issues](https://github.com/PaddlePaddle/PaddleOCR/issues?page=1&q=is%3Aopen+is%3Aissue+label%3Atriaged)。

### 认领方式

请大家以 comment 的形式认领任务，如：

```
【报名】：1、3、12-13
```

多个任务之间需要使用**中文顿号**分隔，报名多个连续任务可用横线表示，如 2-5
PR 提交格式：在 PR 的标题中以 **【OCR Issue No.】** 开头，注明任务编号

## 看板信息
| 任务方向 | 任务数量 | 提交作品 / 任务认领 | 提交率 | 完成 | 完成率 |
| :----: | :----: | :----:  | :----: | :----: | :----: |
|  Fundable Projects No.6  |  12  | 2 / 4 | 16.67% |  2  | 16.67% |
#####

## 统计信息 
> 排名不分先后 @mattheliu (1) @Liyulingyue (1) 
#####

**Comment by GreatV** [comment]
【报名】：5

**Comment by mattheliu** [comment]
【报名】：2、8

**Comment by Liyulingyue** [comment]
【报名】：9、12
